### ISSUE TITLE
assert_type evaluates expressions within func scope as `Any` 

### ISSUE THREAD

**Issue by rsokl** [python/mypy#13624]
**Bug Report**

Using `assert_type` within a function's scope produces unexpected behavior: it evaluates expressions to be `Any` inappropriately.

**Edit:** It looks like `reveal_type` also has this behavior!

```python

```

**To Reproduce**

```python
# global_scope.py
from typing_extensions import assert_type

def f(x: int) -> str: ...

assert_type(f(1), str)
```
```console
$ mypy global_scope.py
Success: no issues found in 1 source file
```
```python
# func_scope.py
from typing_extensions import assert_type

def scoped():
    def f(x: int) -> str: ...

    assert_type(f(1), str)
```
```console
$ mypy func_scope.py 
func_scope.py:7: error: Expression is of type "Any", not "str"
Found 1 error in 1 file (checked 1 source file)
```
**Expected Behavior**

I would expect mypy to report no errors for `func_scope.py` in addition to `global_scope.py`.

**Your Environment**

- Mypy version used: mypy 0.971 (compiled: yes)
- Mypy command-line flags: None
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: 3.9.13
- Operating system and version: Windows 11

**Comment by rsokl** [comment]
The issue here was that `scoped` was an unchecked function. Doing

```python
# func_scope.py
from typing_extensions import assert_type

def scoped() -> None:
    def f(x: int) -> str: ...

    assert_type(f(1), str)
```

fixes this.
